templates/textcat_prompt.jinja2

```diff
@@ -0,0 +1,38 @@
+{% if labels|length == 1 %}
```

Comment:
I may have stumbled apon an improvement here, but feel free to challenge me on this. 

I ran this command; 

```
python -m prodigy textcat.openai.correct another-demo-man example.jsonl templates/textcat_prompt.jinja2 --label "website purchase,complaint,delivery related,compliment" --batch-size 10 --examples-path examples.yaml --max-examples 3 --verbose -F recipes/openai_textcat.py
```

These are the examples that I passed. 

```yaml
- text: "The delivery was great. This has been great. Thanks! https://twitter.com"
  answer: "compliment,delivery related"
  reason: "This is a very general compliment about a delivery. It's not about the website, even though a website appears."
- text: "Good grief. I tried ordering something from the website and it failed."
  answer: "website purchase,complaint"
  reason: "This is complaint about an order that faired on the website."
- text: "Tesco are complaints taken seriously? Do you take customer complaints seriously or just pay it lip service???"
  answer: "complaint"
  reason: "This is complaint but it's about customer service, not about deliveries."
```

What I'm trying to test here, is if it's able to correctly predict a sentence when it's part of the examples listed in the prompt. 

It ended up generating this prompt to send to OpenAI. 

```
Classify the text below to any of the following labels: website purchase, complaint, delivery related, compliment

The task is non-exclusive, so you can provide more than one label as long as
they're comma-delimited. For example: Label1, Label2, Label3.



Your answer should only be in the following format:

answer: <string>
reason: <string>

Text:
"""
Tesco are complaints taken seriously? Do you take customer complaints seriously or just pay it lip service???
"""


Below are some examples (only use these as a guide):



Text:
"""
The delivery was great. This has been great. Thanks! https://twitter.com
"""

answer: compliment,delivery related
reason: This is a very general compliment about a delivery. It's not about the website, even though a website appears.

Text:
"""
Good grief. I tried ordering something from the website and it failed.
"""

answer: website purchase,complaint
reason: This is complaint about an order that faired on the website.

Text:
"""
Tesco are complaints taken seriously? Do you take customer complaints seriously or just pay it lip service???
"""

answer: complaint
reason: This is complaint but it's about customer service, not about deliveries.
```

I copied this prompt and gave it to the OpenAI playground to see what would happen. 

![CleanShot 2023-01-24 at 12 44 11](https://user-images.githubusercontent.com/1019791/214283395-4a873bbd-3827-42d6-a0af-8a6fa3152e28.png)

It had trouble returning anything and I imagine that it's because of the order of items in the prompt. On a high level, the prompt communicates the following order:

```
1. TASK DESCRIPTION
2. THING TO CLASSIFY
3. EXAMPLES OF CLASSIFICATIONS
```

So I tried a different order. 

```
1. TASK DESCRIPTION
2. EXAMPLES OF CLASSIFICATIONS
3. THING TO CLASSIFY
```

That means that this becomes the new prompt:

```
Classify the text below to any of the following labels: website purchase, complaint, delivery related, compliment

The task is non-exclusive, so you can provide more than one label as long as
they're comma-delimited. For example: Label1, Label2, Label3.

Your answer should only be in the following format:

answer: <string>
reason: <string>

Below are some examples (only use these as a guide):

Text:
"""
The delivery was great. This has been great. Thanks! https://twitter.com
"""

answer: compliment,delivery related
reason: This is a very general compliment about a delivery. It's not about the website, even though a website appears.

Text:
"""
Good grief. I tried ordering something from the website and it failed.
"""

answer: website purchase,complaint
reason: This is complaint about an order that faired on the website.

Text:
"""
Tesco are complaints taken seriously? Do you take customer complaints seriously or just pay it lip service???
"""

answer: complaint
reason: This is complaint but it's about customer service, not about deliveries.

Here is the text that needs classification:

Text:
"""
Tesco are complaints taken seriously? Do you take customer complaints seriously or just pay it lip service???
"""

answer:
```

When I tried that, it seems to return what I am interested in. 

![CleanShot 2023-01-24 at 12 41 17](https://user-images.githubusercontent.com/1019791/214282993-ff0c1cbb-f053-4193-9f40-ed03b9b1bf51.png)

I want to be careful with making _another_ suggestion to the prompt. I also need to accept that we shouldn't expect too much from OpenAI. But this feels like a simple enough change that makes a positive difference. At least on my dataset.

Reply:
Yeah I also realized that prompting it that way helps. I'll test this tomorrow then update the prompt 👍

Reply:
Seems to work with no regression, will replace the current prompt with this one.